### PR TITLE
Issue 3 user can edit an entry

### PR DESCRIPTION
**Pull request by KiraHike**
image
image
image
image

### CODE REVIEW

File: js/main.js [hunk 1]
```diff
@@ -25,37 +26,75 @@ var $newEntryForm = document.querySelector('#entry-form');
...
+    titleValue = $newEntryForm.elements.title.value;
+    notesValue = $newEntryForm.elements.notes.value;
+    imageValue = $newEntryForm.elements.url.value;
+    entryObj = {
+      url: imageValue,
+      title: titleValue,
+      notes: notesValue,
+      entryId: data.nextEntryId
+    };
+    event.preventDefault();
```
Comment (codingcodymiller) [inline js/main.js]: You should make this the first line of the function

File: js/main.js [hunk 2]
```diff
@@ -25,37 +26,75 @@ var $newEntryForm = document.querySelector('#entry-form');
...
+    data.nextEntryId++;
+    data.entries.unshift(entryObj);
+    $entry = renderEntry(entryObj);
+    $entryList.prepend($entry);
+    $image.setAttribute('src', 'images/placeholder-image-square.jpg');
+    $newEntryForm.reset();
+    changeViewEntries();
+  } else {
+    event.preventDefault();
```
Comment (codingcodymiller) [inline js/main.js]: Which will remove the need for this line

File: js/main.js [hunk 3]
```diff
@@ -25,37 +26,75 @@ var $newEntryForm = document.querySelector('#entry-form');
...
+
+var titleValue = $newEntryForm.elements.title.value;
+var notesValue = $newEntryForm.elements.notes.value;
+var imageValue = $newEntryForm.elements.url.value;
+var entryObj = {
+  url: imageValue,
+  title: titleValue,
+  notes: notesValue,
+  entryId: data.nextEntryId
+};
```
Comment (codingcodymiller) [inline js/main.js]: This is in the global scope, I think it may not be supposed to be here.

**Review by codingcodymiller — CHANGES_REQUESTED** [review]
Great work so far!! Please see my notes

File: js/main.js (hunk 1 above)
Comment (KiraHike) [reply js/main.js]: Done.

File: js/main.js (hunk 2 above)
Comment (KiraHike) [reply js/main.js]: Done.

File: js/main.js (hunk 3 above)
Comment (KiraHike) [reply js/main.js]: Moved to clickSave function.

**Comment by KiraHike** [comment]
Made requested revisions

File: js/main.js [hunk 4]
```diff
@@ -83,7 +124,7 @@ function renderEntry(object) {
 }
 
 function contentLoaded(event) {
-  for (var i = 0; i < data.entries.length; i++) {
+  for (i = 0; i < data.entries.length; i++) {
```
Comment (thebearingedge) [inline js/main.js]: you should definitely declare your variable here instead of using a global.

File: js/main.js [hunk 5]
```diff
@@ -98,7 +139,7 @@ function changeViewNav(event) {
   if (event.target.matches('.nav-link')) {
     var dataViewValue = event.target.getAttribute('data-view');
     data.view = dataViewValue;
-    for (var i = 0; i < $viewList.length; i++) {
+    for (i = 0; i < $viewList.length; i++) {
```
Comment (thebearingedge) [inline js/main.js]: you should definitely declare your variable here instead of using a global.

File: js/main.js [hunk 6]
```diff
@@ -109,3 +150,30 @@ function changeViewNav(event) {
...
+var dataEntryIDValue;
+
+var $formHead = document.querySelector('h2');
+var dataEntriesIndex;
+
+function editEntry(event) {
+  if (event.target.matches('i')) {
+    closestElement = event.target.closest('li');
+    dataEntryIDValue = Number(closestElement.getAttribute('data-entry-id'));
+    for (i = 0; i < data.entries.length; i++) {
```
Comment (thebearingedge) [inline js/main.js]: you should definitely declare your variable here instead of using a global.

**Review by thebearingedge — CHANGES_REQUESTED** [review]
Very nice, just a couple of small changes.

File: js/main.js (hunk 4 above)
Comment (KiraHike) [reply js/main.js]: Done.

File: js/main.js (hunk 5 above)
Comment (KiraHike) [reply js/main.js]: Done.

File: js/main.js (hunk 6 above)
Comment (KiraHike) [reply js/main.js]: Done.

**Comment by KiraHike** [comment]
Made requested revisions.

**Review by codingcodymiller — APPROVED** [review]
🥳 🥳 🥳 🥳

**Pull request merged**